packages/asset-swapper/src/rpc_sampler_client.ts

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+
+import {
+    Address,
+    LiquidityResponse,
+    RpcLiquidityRequest,
+    TokenResponse,
+} from './utils/market_operation_utils/sampler_types';
+
+const SAMPLER_SERVICE_URL = '';
+
+export interface SamplerServiceInterface {
+    getSellLiquidityAsync(reqs: RpcLiquidityRequest[]): Promise<LiquidityResponse[]>;
+    getBuyLiquidityAsync(reqs: RpcLiquidityRequest[]): Promise<LiquidityResponse[]>;
+    getTokensAsync(tokenAddresses: Address[]): Promise<TokenResponse[]>;
+    // getPrices(tokenPaths: Address[][]): Promise<bigint[]>;
+}
+
+export class RpcSamplerClient implements SamplerServiceInterface {
+    // tslint:disable:prefer-function-over-method
+    public async getSellLiquidityAsync(reqs: RpcLiquidityRequest[]): Promise<LiquidityResponse[]> {
+        try {
+            return await axios.post(`${SAMPLER_SERVICE_URL}/get_sell_liquidity`, reqs, {
```

Comment:
The server is in JSONRPC, so this will just 400 you. You need a JSONRPC client library. It'll also abstract away all the header boilerplate.

Reply:
And don't forget to convert `BigNumber` into string for the params.